### ISSUE TITLE
validateArn incorrectly rejects ARNs in isolated AWS regions

### ISSUE THREAD

**Issue by ccday** [terraform-providers/terraform-provider-aws#8307]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.13
+ provider.aws v2.6.0

### Affected Resource(s)

* All resources that have a field that uses `validateArn` for validation

### Terraform Configuration Files

```hcl
resource "aws_ebs_volume" "test" {
  // Some configuration...

  encrypted = true
  kms_key_id = "arn:aws-iso:kms:us-iso-east-1:012345678901:key/some-uuid-abc123"
}
```

### Expected Behavior

The ARN should validate.

### Actual Behavior

The ARN does not validate:
```
aws_ebs_volume.test: "kms_key_id" doesn't look like a valid ARN ("^arn:[\\w-]+:([a-zA-Z0-9\\-])+:([a-z]{2}-(gov-)?[a-z]+-\\d{1})?:(\\d{12})?:(.*)$"): <arn here>
```

Suggested fix is to update the regex in the `validateArn` function of `aws/validators.go` to match on ARNs from the iso and isob partitions:

```go
pattern := `^arn:[\w-]+:([a-zA-Z0-9\-])+:([a-z]{2}-((?:gov|iso|isob)-)?[a-z]+-\d{1})?:(\d{12})?:(.*)$`
```

### Steps to Reproduce

1. Create a configuration that uses an ARN from an isolated AWS region.
2. `terraform validate`

### Important Factoids

This account is located in an isolated AWS region.

**Comment by ccday** [comment]
I'd be happy to make a PR with a fix for this issue - let me know if the suggested fix sounds good.

**Comment by ewbankkit** [comment]
Related:
* https://github.com/terraform-providers/terraform-provider-aws/issues/9993

**Comment by ewbankkit** [comment]
Probably most sustainable is to use the AWS SDK's [`arn.Parse`](https://docs.aws.amazon.com/sdk-for-go/api/aws/arn/#Parse) method.
It doesn't validate quite as extensively the regex but its the regex that is brittle and it will certainly catch the most common types of errors which are probably usage of a resource ID or name instead of an ARN.

**Comment by ewbankkit** [comment]
Similar:
* https://github.com/terraform-providers/terraform-provider-aws/issues/11444

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.44.0 of the Terraform AWS Provider, on Thursday. 👍
